lib/sycamore/sycamore/utils/bbox_sort.py

```diff
@@ -0,0 +1,151 @@
+"""
+Utilities to sort elements based on bounding box (bbox) coordinates.
+
+TODO:
+- handle page_number not (always) present
+- handle bbox not (always) present
+"""
+
+from typing import Optional
+
+from sycamore.data import Document, Element
+
+
+def elem_top_left(elem: Element) -> tuple:
+    bbox = elem.data.get("bbox")
+    if bbox:
+        return (bbox[1], bbox[0])
+    return (0.0, 0.0)
+
+
+def elem_left_top(elem: Element) -> tuple:
+    bbox = elem.data.get("bbox")
+    if bbox:
+        left = int(5 * bbox[0])  # !!! quantize
+        return (left, bbox[1])
+    return (0.0, 0.0)
+
+
+def collect_pages(elems: list[Element]) -> list[list[Element]]:
+    """
+    Collect elements into page-number buckets.  Basically like the first
+    stage of a radix sort.
+    """
+    pagemap: dict[int, list[Element]] = {}
+    for elem in elems:
+        page = elem.properties.get("page_number", 0)
+        ary = pagemap.get(page)
+        if ary:
+            ary.append(elem)
+        else:
+            pagemap[page] = [elem]
+    nums = list(pagemap.keys())
+    nums.sort()
+    rv = []
+    for page in nums:
+        ary = pagemap[page]
+        rv.append(ary)
+    return rv
+
+
+def col_tag(elem: Element) -> Optional[str]:
+    bbox = elem.data.get("bbox")
+    if bbox:
+        left = bbox[0]
+        right = bbox[2]
+        width = right - left
+        if width > 0.6:
+            return "full"
+        elif (width < 0.1) or (width >= 0.45):
+            return None
+        if right < 0.5:
+            return "left"
+        elif left > 0.5:
+            return "right"
+    return None
+
+
+def find_overlap(top: float, bot: float, elems: list[Element]) -> list[Element]:
+    """
+    Assumes elems is sorted by top, ascending.
```

Comment:
Maybe add to the docstring: Finds elements from list that fall between the y-coordinates `top` and `bot`.